files-csharp/FastestWayToReadATextFileInCsharp/FastestWayToReadATextFileInCsharp/WaysToReadATextFileInCsharpBenchmark.cs

```diff
@@ -0,0 +1,54 @@
+﻿using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Order;
+
+namespace FastestWayToReadATextFileInCsharp;
+
+[MemoryDiagnoser]
+[Orderer(SummaryOrderPolicy.FastestToSlowest)]
+public class WaysToReadATextFileInCsharpBenchmark
+{
+    private static readonly string SampleFilePath
+       = Path.Combine(Directory.GetParent(Directory.GetCurrentDirectory()).Parent.Parent.Parent.Parent.Parent.Parent.FullName, "BenchmarkTextFile.txt");
```

Comment:
Is this needed, why can't we just put the name of the file "BenchmarkTextFile.txt" in the variable, and include the file in the build folder?

Reply:
Whenever I try this, I get the FileNotFoundException with this exception message:

`System.IO.FileNotFoundException: Could not find file 'C:\Users\Januarius\Desktop\CodeMazeGuides\files-csharp\FastestWayToReadATextFileInCsharp\FastestWayToReadATextFileInCsharp\bin\Release\net7.0\29df687e-8207-4cf8-970a-abac7154797d\bin\Release\net7.0\BenchmarkTextFile.txt'`

From this message, the benchmark file should be in that build folder, which I cannot locate on my PC.
It also means I have to push the bin\Release folder too.

I followed this approach because it was suggested by Marinko in a previous article I worked on: https://github.com/CodeMazeBlog/CodeMazeGuides/blob/main/json-csharp/ReadAndParseAJSONFileInCSharp/ReadAndParseAJSONFileInCSharp/ReadAndParseJSONFileMethodsBenchMark.cs

Reply:
@cjaynjoku and @karimkod - another potential option would be to use the `[GlobalSetup]` and `[GlobalCleanup]` methods and write the text out to a temporary file. We could create the file in the setup method and delete it in the cleanup method.

Reply:
I have chosen the option above and it works perfectly. This article has dragged on for over a month. Thank you.